Full code pull request

TheCoolerSuptelov opened the pull request:
Не успел доделать докер файлы, спринг не видит pgSql. Если подскажешь раньше чем исправлю буду супер благодарен.

Вопрос: можно ли вытащить "шаблон" сущности из бд. Допустим, у нас есть реквизиты, которые мы можем достать из БД, но самой сущности нет. Чтобы не плодить запросы, хотелось бы вытащить все значения реквизитов, а потом просто записать объект и назначить ссылку. Есть выкрутасы через ResultSet, и работу с Entity manager, но мне кажется, что должно быть решение проще.

TheCoolerSuptelov committed:
Add default server port
	modified:   src/main/resources/application.properties

TheCoolerSuptelov committed:
Table creation.
Class creation.
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	new file:   src/main/resources/db/migration/V1__initSchema.sql
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql

TheCoolerSuptelov committed:
Add service class
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql

TheCoolerSuptelov committed:
Add repo classes
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql

TheCoolerSuptelov committed:
Add @CreationTimestamp
Add User creatin function in controller
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Fixed Sonar report issues
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Modif checkStyle issues
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplicationTests.java

TheCoolerSuptelov committed:
Add user test. Change schema and hibernate entity naming strategy
	modified:   pom.xml
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplicationTests.java
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java

TheCoolerSuptelov committed:
Made the field errorDetails optional on json output
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java

TheCoolerSuptelov committed:
Test improvements
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java

TheCoolerSuptelov committed:
Wrote integration test on user api
	modified:   pom.xml
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java
	new file:   src/test/resources/application.properties
	new file:   src/test/resources/docker-compose.yml

TheCoolerSuptelov committed:
Done rest api chat creation

	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java

TheCoolerSuptelov committed:
Fix table "Chat" structure. End with creation chat using list userNames
	modified:   README.md
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	new file:   src/main/resources/db/migration/V2__AddCreationAtToChat.sql

TheCoolerSuptelov committed:
Edite response chat and user controller
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java

TheCoolerSuptelov committed:
Failed to find chat and user in 1 query
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Add message controller logic.
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Rest Api method that return all chats concrete user.
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/resources/application.properties

TheCoolerSuptelov committed:
Add rest api method to get all messages from 1 paticular chat.
	modified:   README.md
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Reorganize files
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java

TheCoolerSuptelov committed:
Refactor UserDto
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplicationTests.java

TheCoolerSuptelov committed:
Refactor MessageDto
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Refactor ChatDto
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Add pagination
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Add docker-compose for test environment
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java
	modified:   src/test/resources/application.properties
	modified:   src/test/resources/docker-compose.yml

TheCoolerSuptelov committed:
Added dockerFiles
	new file:   Dockerfile
	new file:   docker-compose.yml

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java:
```diff
@@ -0,0 +1,57 @@
+        this.id = id;
+    }
+
+}
```
Этой красной штуки не должно быть.

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java:
```diff
@@ -0,0 +1,50 @@
+        String processedChat;
+        if (findBy != null && !findBy.isEmpty()){
+           processedChat = chatService.addNewChat(chatFromRequest, findBy);
+        }else {
```
форматирование мелочь, а не приятно.

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java:
```diff
@@ -0,0 +1,50 @@
+    }
+    @PostMapping(value = "add")
+    public ResponseEntity<String> createChatWithUsers(@RequestBody ChatDto chatFromRequest, String findBy) {
+        // Убрать кишки чат сервис. Сделать 1 функции внутри сервиса.
```
да, никакой бизнес логики на уровне представления!

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+    boolean isUserInAChat(ChatDto chatFromRequest) {
+        return getChatRepository()
+                .findByNameEqualsIgnoreCase(
+                        chatFromRequest.getName()
```
зачем такое форматирование?

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+                )
+                .isPresent();
+    }
+    public String addNewChat(ChatDto chatFromRequest, String userSearchAttribute) {
```
FromRequest - какая смысловая нагрузка тут? зачем она?

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+            usersInChat = userService.getUserRepository().findByUsernameIn(chatFromRequest.getUsers());
+        } else {
+            usersInChat = userService.getUserRepository().findByIdIn(
+                    chatFromRequest
```
Я бы этот результат засунул в var переменную и передал в метод.

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+        this.userService = userService;
+    }
+
+    public ChatRepository getChatRepository() {
```
Зачем это и что это? бины не надо шарить между бинами.

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+        this.chatRepository = chatRepository;
+    }
+    boolean isUserInAChat(ChatDto chatFromRequest) {
+        return getChatRepository()
```
а) почему у метода дефолтный досутп? б) зачем вызов этого метода?

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+        // TODO
+        // use Strategy pattern
+        // move to userService
+        if (userSearchAttribute == "username") {
```
сильно жесткий хардкод.

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+                .map(e -> e.getUsername())
+                .collect(Collectors.joining(","));
+
+        return "Cannot create chat.Users not found: " + notFoundedUsernames;
```
никогда так не делай. Если произошла ошибка - бросаем исключение и где надо ловим!

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java:
```diff
@@ -0,0 +1,89 @@
+        if (userSearchAttribute == "username") {
+            usersInChat = userService.getUserRepository().findByUsernameIn(chatFromRequest.getUsers());
+        } else {
+            usersInChat = userService.getUserRepository().findByIdIn(
```
<img width="1152" alt="image" src="https://user-images.githubusercontent.com/2914310/177054476-320a100b-841d-4125-aae1-4acd6d22a390.png">
Нижние слои не должны знать о верхних, а слои одного уровня по хорошему не должны знать друг о друге и жесткой связности между ними не должно быть.

AndryLat replied:
<img width="1141" alt="image" src="https://user-images.githubusercontent.com/2914310/177054525-ee182617-86f1-4616-9bc7-5e4416d4a7f0.png">

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java:
```diff
@@ -0,0 +1,41 @@
+
+    @PostMapping(value = "get")
+    public ResponseEntity<List<MessageDto>> getAllMessagesFromChat(@RequestBody ChatDto chatDto) {
+        var xx = messageService.getAllMessagesFromChat(chatDto.getId());
```
xx? серьезно?

AndryLat commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Ломбок аж просится сюда:)

AndryLat commented on src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java:
```diff
@@ -0,0 +1,64 @@
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class TestUserApi {
```
Явно не все покрывает. Даже если брать с точки зрения интеграционности.

TheCoolerSuptelov committed:
Reorganize chatService
	modified:   Dockerfile
	modified:   docker-compose.yml
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java
	modified:   src/test/resources/docker-compose.yml

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Refactored unnecessary dependencies between services
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Add exception. Not analyze input on services level, raise error instead.

	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Rename variables. Delete "fromRequest"
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java

TheCoolerSuptelov committed:
Change error handling in chatService. Instead of string with error description, throw errors
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Change logic user search. Instead of hardCoded attribute names, use ENUM
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java

TheCoolerSuptelov committed:
UserService now throw error then user trying create username with existed name.
Renamed test in TestUserApi.
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java

TheCoolerSuptelov committed:
Fix the ChatControllerTest.
Add flyway sql script test migration.
	new file:   src/main/resources/testdata/afterMigrate.sql
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	renamed:    src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java -> src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java
	modified:   src/test/resources/application.properties

TheCoolerSuptelov committed:
Formatting
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java
	modified:   src/test/resources/application.properties

TheCoolerSuptelov committed:
Add message test cases
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/resources/testdata/afterMigrate.sql
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java
	modified:   src/test/resources/application.properties

TheCoolerSuptelov committed:
Fix sonar issues. Reformat code
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplication.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java

TheCoolerSuptelov committed:
Delete setters ID for Entities
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/EndToEndApiTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java

TheCoolerSuptelov committed:
Add end to end test
    modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/EndToEndApiTest.java

TheCoolerSuptelov committed:
Add test to chat creation with invalid user UUID
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java

TheCoolerSuptelov committed:
Reformat code
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
